app/ai/voice/stt/soniox/service.py

```
@@ -0,0 +1,94 @@
+            if self._websocket and self._websocket.state is State.OPEN:
+                return
+
+            logger.debug("Connecting to Soniox STT (enhanced)")
+
+            self._websocket = await websocket_connect(self._url)
+
+            if not self._websocket:
+                await self.push_error(
+                    error_msg=f"Unable to connect to Soniox API at {self._url}"
+                )
+                raise Exception(f"Unable to connect to Soniox API at {self._url}")
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)